cfn/encoding/marshal.go

```diff
@@ -0,0 +1,31 @@
+package encoding
+
+import (
+	"encoding/json"
+)
+
+// Marshal converts a value into stringified-JSON
```

Comment:
It's not just weird, it's downright horrible. It's really just to avoid having to deal with all sorts of types in Stringify. This way, we can use a roundtrip in the json marshaller to guarantee we'll only ever get maps, slices, and scalars. Otherwise, we'd have to account for pointers and structs too (which would require a *lot* more code in Stringify).

Reply:
Actually, I decided it wasn't all *that* much work and have added it. Quick review of stringify please :)